src/controllers/tasks/natural.person.secretaries.controller.ts

```diff
@@ -31,9 +31,31 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
     const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
     const session: Session = req.session as Session;
+    const officers: ActiveOfficerDetails[] = await getActiveOfficersDetailsData(session, transactionId, submissionId);
     const natPersonSecretariesBtnValue = req.body.naturalPersonSecretaries;
+
     if (natPersonSecretariesBtnValue === RADIO_BUTTON_VALUE.YES || natPersonSecretariesBtnValue === RADIO_BUTTON_VALUE.RECENTLY_FILED) {
-      // TODO update with correct next page in journey (not task list)
+      let corpSecretary = false;
```

Comment:
for the sonar code smell for this method you can split the if-else if -else up
the elseif part i would turn into a 'guard clause' which you can do early in this function
you could split them into if-then-return blocks like this without using 'else'

`if (natPersonSecretariesBtnValue === RADIO_BUTTON_VALUE.NO) {
      return res.render(Templates.WRONG_DETAILS, {
        templateName: Templates.WRONG_DETAILS,
        backLinkUrl: urlUtils.getUrlToPath(NATURAL_PERSON_SECRETARIES_PATH, req),
        returnToTaskListUrl: urlUtils.getUrlToPath(TASK_LIST_PATH, req),
        stepOneHeading: WRONG_DETAILS_UPDATE_SECRETARY,
        pageHeading: WRONG_DETAILS_UPDATE_OFFICERS,
      });`

`if (natPersonSecretariesBtnValue === RADIO_BUTTON_VALUE.YES || natPersonSecretariesBtnValue === RADIO_BUTTON_VALUE.RECENTLY_FILED) {
      let corpSecretary = false;
      let director = false;
      let corpDirector = false;
      officers.forEach(officer => {
        if (officer.role === "SECRETARY" && officer.isCorporate){
          corpSecretary = true;
        } else if (officer.role === "DIRECTOR" && !officer.isCorporate){
          director = true;
        } else if (officer.role === "DIRECTOR" && officer.isCorporate){
          corpDirector = true;
        }
      });

      if (corpSecretary){
        return res.redirect(urlUtils.getUrlToPath(CORPORATE_SECRETARIES_PATH, req));
      } else if (director){
        return res.redirect(urlUtils.getUrlToPath(NATURAL_PERSON_DIRECTORS_PATH, req));
      } else if (corpDirector){
        return res.redirect(urlUtils.getUrlToPath(CORPORATE_DIRECTORS_PATH, req));
      }

      return res.redirect(urlUtils.getUrlToPath(TASK_LIST_PATH, req));
    }`

then the final part wouldn't need to be wrapped in an else as the other 'ifs' above would return if they were entered

`const officers: ActiveOfficerDetails[] = await getActiveOfficersDetailsData(session, transactionId, submissionId);
      const secretaryList = formatSecretaryList(officers);
      return res.render(Templates.NATURAL_PERSON_SECRETARIES, {
        backLinkUrl: urlUtils.getUrlToPath(TASK_LIST_PATH, req),
        secretaryErrorMsg: SECRETARY_DETAILS_ERROR,
        templateName: Templates.NATURAL_PERSON_SECRETARIES,
        secretaryList
      });`

Reply:
formatting hasn't come out very well in my comment sorry

Reply:
hopefully that should reduce the sonar complexity rating